pilot/pkg/networking/core/v1alpha3/envoyfilter.go

```diff
@@ -246,7 +246,7 @@ func insertHTTPFilter(listenerName string, filterChain *listener.FilterChain, hc
 	// Its the last filter in the filter chain
 	filterChain.Filters[len(filterChain.Filters)-1] = listener.Filter{
 		Name:       xdsutil.HTTPConnectionManager,
-		ConfigType: &listener.Filter_Config{Config: util.MessageToStruct(hcm)},
+		ConfigType: &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(hcm)},
```

Comment:
what about the other filers in the system added through EnvoyFilters?

Reply:
both parties should know proto schema in that case, so there is no problem. is there some negotiation going on about enabled filters? I haven't seen anything like that.

Reply:
what I meant is that this is simply marshalling only HCM. EnvoyFilters allows you to add both network filters and HCM filters.

Reply:
I'm not familiar with envoy filters. I did a mechanical substitute of all struct conversions. I haven't found any other use of struct after this change.